finals-2024-01-16/code/5130-stefan-vitanov/src/app/band-form/band-form.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import Band, { defaultBand } from '../models/band';
+import { PlacesService } from '../services/places.service';
+
+@Component({
+  selector: 'app-band-form',
+  templateUrl: './band-form.component.html',
+  styleUrl: './band-form.component.css'
+})
+export class BandFormComponent implements OnInit{
+
+  constructor(private _placesService: PlacesService) {}
+
+  countries = [""];
+  cities = [""];
+
+  @Input()
+  band: Band = defaultBand();
+
+  @Output()
+  cancel = new EventEmitter<void>();
+
+  @Output()
+  save = new EventEmitter<{name: string, formed: number, country: string, city: string, genre: string}>();
+
+  bandForm = new FormGroup({
```

Comment:
This could have used the `defaultBand` to fill the data. Anyway, +5 points for using `FormGroup``